src/main/java/org/gradle/profiler/mutations/ClearProjectCacheMutator.java

```diff
@@ -0,0 +1,57 @@
+package org.gradle.profiler.mutations;
+
+import com.typesafe.config.Config;
+import org.apache.commons.io.FileUtils;
+import org.gradle.profiler.BuildMutator;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.function.Supplier;
+
+public class ClearProjectCacheMutator implements BuildMutator {
+    private final File projectDir;
+
+    public ClearProjectCacheMutator(File projectDir) {
+        this.projectDir = projectDir;
+    }
+
+    @Override
+    public void afterBuild(Throwable error) {
+        deleteGradleCache("project", projectDir);
+        File buildSrc = new File(projectDir, "buildSrc");
+        if (buildSrc.exists()) {
+            deleteGradleCache("buildSrc", buildSrc);
+        }
+    }
+
+    private void deleteGradleCache(String name, File baseDir) {
+        File gradleCache = new File(baseDir, ".gradle");
+        System.out.println(String.format("> Cleaning %s .gradle cache: %s", name, gradleCache));
+        if (!gradleCache.exists()) {
+            throw new IllegalArgumentException(String.format(
+                "Cannot delete .gradle %s directory (%s) since it does not exist",
+                name,
+                gradleCache
+            ));
+        }
+        try {
+            FileUtils.deleteDirectory(gradleCache);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    @Override
+    public String toString() {
+        return getClass().getSimpleName();
+    }
```

Comment:
This should be consistent, but the other mutator doesn't add it. Shall we make it the default?

Reply:
I added a new base class which makes it the default. It can't be a default method on the interface, since the method is defined in `Object`...